src/main/java/org/example/Main.java

```diff
@@ -0,0 +1,85 @@
+package org.example;
+
+import org.example.course.Course;
+import org.example.department.Department;
+import org.example.student.Student;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class Main {
+    public static void main(String[] args) {
+        // ************************************************************************
+        // Data initialization
+        List<Student> students = new ArrayList<>(List.of(
+                new Student("A", "Aa", "0000", 20, Student.Gender.MALE),
+                new Student("B", "Bb", "0001", 26, Student.Gender.FEMALE),
+                new Student("C", "Cc", "0002", 19, Student.Gender.MALE),
+                new Student("D", "Dd", "0003", 25, Student.Gender.FEMALE),
+                new Student("E", "Ee", "0004", 20, Student.Gender.MALE),
+                new Student("F", "Ff", "0005", 24, Student.Gender.MALE),
+                new Student("G", "Gg", "0006", 19, Student.Gender.FEMALE),
+                new Student("H", "Hh", "0007", 23, Student.Gender.MALE),
+                new Student("I", "Ii", "0008", 21, Student.Gender.FEMALE),
+                new Student("J", "Jj", "0009", 22, Student.Gender.FEMALE)
+        ));
+        List<Department> departments = new ArrayList<>(List.of(
```

Comment:
Those assignments are a bit too complicated - in a real production application that wouldn't be a good approach
but for a task like that it can be ;)